tests/common.rs

```diff
@@ -54,9 +54,10 @@ pub async fn wait_n_finalized_blocks(
 async fn wait_n_finalized_blocks_from(n: usize, url: &str) {
 	let mut built_blocks = std::collections::HashSet::new();
 	let mut interval = tokio::time::interval(Duration::from_secs(6));
+	let rpc_service = RpcService::new(url, false).await.unwrap();
```

Comment:
This `.unwrap()` looks dangerous. Previously errors were handled by `if let` below: can't we do lazy initialization in the loop to try to recover from them? CC @niklasad1

Reply:
initially, I put `unwrap` here since afaik this method serves just for testing purposes; anyway, I modified it - does it look better now?

Reply:
yeah sure, that's a good point

Reply:
either way is fine for me, 

The rpc connections is only used for `CLI integrations tests` and if the connection fails means that RPC server hasn't started or the URL is wrong. 

Now we try to connect forever as it was before so fine which would be caught by the CI :)